packages/widget-chat/src/containers/notifications/index.js

```diff
@@ -0,0 +1,125 @@
+import {Component} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import _ from 'lodash';
+
+import {getUnsentNotifications} from '../../selectors/notifications';
+import {notificationSent, setNotificationPermission, setNotificationSupported} from '../../actions/notifications';
+
+const TIMEOUT_LENGTH = 10000;
+
+class Notifications extends Component {
+  shouldComponentUpdate(nextProps) {
+    const {props} = this;
+    return nextProps.notifications !== props.notifications || nextProps.isSupported !== props.isSupported || nextProps.permission !== props.permission;
+  }
+
+  componentDidUpdate() {
+    const {props} = this;
+    // Default state is not supported, once we know it is, don't check any more
+    if (!props.isSupported) {
+      this._checkSupported();
+    }
+    if (_.isNull(props.permission) && props.isSupported) {
+      Notification.requestPermission((permission) => props.setNotificationPermission(permission));
+    }
+    this._displayNotifications();
+  }
+
+  /**
+   * Checks if browser notifications are supported and updates store
+   *
+   * @returns {null}
+   */
+  _checkSupported() {
+    const {props} = this;
+    if (!props.isSupported && window && window.Notification) {
+      props.setNotificationSupported(true);
+    }
+  }
+
+  /**
+   * Checks if we have permission from the user to send a notification
+   *
+   * @returns {bool}
+   *
+   * @memberOf Notifications
+   */
+  _hasPermission() {
+    const {props} = this;
+    return props.permission === `granted`;
+  }
+
+  /**
+   * Returns true if the main browser window is hidden, minimized,
+   * or in the background; false otherwise.
+   *
+   * @returns {boolean} hidden
+   */
+  _isBrowserHidden() {
+    return document.webkitHidden || document.mozHidden || document.msHidden || document.hidden || !document.hasFocus();
+  }
+
+  /**
+   * Processes notifications and displays them if needed
+   *
+   * @returns {null}
+   */
+  _displayNotifications() {
+    const {props} = this;
+    if (props.notifications.length > 0) {
```

Comment:
if visible notifications were tied to `<Notification>` elements and store entries, there wouldn't be a need to have a private method creating them.